libcontainer/cgroups/fscommon/rdma.go

```diff
@@ -0,0 +1,122 @@
+// +build linux
+
+package fscommon
+
+import (
+	"bufio"
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"golang.org/x/sys/unix"
+)
+
+// parseRdmaKV: parses raw string to RdmaEntry.
+func parseRdmaKV(raw string, entry *cgroups.RdmaEntry) error {
+	var value uint32
+
+	parts := strings.SplitN(raw, "=", 3)
+
+	if len(parts) != 2 {
+		return errors.New("Unable to parse RDMA entry")
+	}
+
+	k, v := parts[0], parts[1]
+
+	if v == "max" {
+		value = math.MaxUint32
+	} else {
+		val64, err := strconv.ParseUint(v, 10, 32)
+		if err != nil {
+			return err
+		}
+		value = uint32(val64)
+	}
+	if k == "hca_handle" {
+		entry.HcaHandles = value
+	} else if k == "hca_object" {
+		entry.HcaObjects = value
+	}
+
+	return nil
+}
+
+// readRdmaEntries: Reads and converts array of rawstrings to RdmaEntries from file.
+func readRdmaEntries(dir, file string) ([]cgroups.RdmaEntry, error) {
+	rdmaEntries := make([]cgroups.RdmaEntry, 0)
+	fd, err := cgroups.OpenFile(dir, file, unix.O_RDONLY)
+	if err != nil {
+		return nil, err
+	}
+	defer fd.Close() //nolint:errorlint
+	scanner := bufio.NewScanner(fd)
+	for scanner.Scan() {
+		parts := strings.SplitN(scanner.Text(), " ", 4)
+		if len(parts) == 3 {
+			entry := new(cgroups.RdmaEntry)
+			entry.Device = parts[0]
+			err = parseRdmaKV(parts[1], entry)
+			if err != nil {
+				continue
+			}
+			err = parseRdmaKV(parts[2], entry)
+			if err != nil {
+				continue
+			}
+
+			rdmaEntries = append(rdmaEntries, *entry)
+		}
+	}
+	return rdmaEntries, scanner.Err()
+}
+
+// RdmaGetStats: Returns rdma stats such as totalLimit and current entries.
+func RdmaGetStats(path string, stats *cgroups.Stats) error {
+	if !cgroups.PathExists(path) {
+		return nil
+	}
+	currentEntries, err := readRdmaEntries(path, "rdma.current")
+	if err != nil {
+		return err
+	}
```

Comment:
It seems that in here we should return all errors except for os.IsNotExist. So, something like

```go
cur, err := readRdmaEntries(path, "rdma.current")
if err != nil && !errors.Is(err, os.IsNotExist) {
    return err
}
```

Now, with this in place, you don't have to check that the path exists beforehand, so you can drop the `if !cgroups.PathExists(path)`.

Reply:
@kolyshkin I think this should be 
```go
	currentEntries, err := readRdmaEntries(path, "rdma.current")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		return err
	}
```
switching `os.IsNotExist` with `os.ErrNotExist` as `os.IsNotExist` is a method which returns `bool`.

Reply:
Yes, that's right, I mistyped.